### PR TITLE
Feature/detailed deliveries api

### PR DESCRIPTION
**Pull request by ptuytsch**
Add new functionallity to the delivery api:
 - Add the delivery history call to get a paginated result of the delivery history
 - Add a query parameter to the parcelbox to nest the delivery result when applicable for the parcelbox.
 - some minor improvements in the controllers

### CODE REVIEW

File: src/gateway/delivery_controller.py [hunk 1]
```diff
@@ -84,9 +75,37 @@ def load_deliveries(user_id=None, include_picked_up=False):
                  (Delivery.user_pickup_id == user_id))
             )
         # filter on picked up when needed
+        query = query.where(Delivery.timestamp_pickup.is_null(not history))
+
+        # add the from_id
+        query = query.where(Delivery.id > from_id)
```
Comment (yclybouw) [inline src/gateway/delivery_controller.py:81]: As the history is shown in descending order (which makes sense), the "after" or "from_id" should be the upper limit, so: `query.where(Delivery.id < from_id)`

**Review by yclybouw — CHANGES_REQUESTED** [review]

File: src/gateway/delivery_controller.py (hunk 1 above)
Comment (yclybouw) [reply src/gateway/delivery_controller.py:81]: This reminds me: from_id should not default to `0` but to "infinity". E.g. set to `None` and do the where conditionally.

Comment (ptuytsch) [reply src/gateway/delivery_controller.py]: indeed good catch! The fix has been pushed

**Review by yclybouw — APPROVED** [review]

**Review by yclybouw — APPROVED** [review]

File: src/gateway/api/V1/parcelbox.py [hunk 2]
```diff
@@ -63,13 +63,17 @@ def __init__(self, esafe_controller=INJECTED, delivery_controller=INJECTED):
                                       controller=self, action='put_parcelbox',
                                       conditions={'method': ['PUT']})
 
-    @openmotics_api_v1(auth=False, expect_body_type=None, check={'size': str, 'available': bool})
-    def get_parcelboxes(self, size=None, available=None):
+    @openmotics_api_v1(auth=False, expect_body_type=None, check={'size': str, 'available': bool, 'show_deliveries': bool})
+    def get_parcelboxes(self, size=None, available=None, show_deliveries=None):
```
Comment (LnL7) [inline src/gateway/api/V1/parcelbox.py:67]: Any user can list pending delivery information?

**Review by LnL7 — DISMISSED** [review]

File: src/gateway/api/V1/parcelbox.py (hunk 2 above)
Comment (ptuytsch) [reply src/gateway/api/V1/parcelbox.py:67]: Good point, will fix this.

**Review by LnL7 — APPROVED** [review]

**Pull request merged**